### PR TITLE
Add CLEAR and FILTER buttons to the notification adapter 

### PR DESCRIPTION
**Pull request by pandasoft0**
Hi

Issues: #762 #764 

I've added a new Top adapter item to the Notifications adapter.
It has 2 buttons: CLEAR and FILTER.
On CLEAR click - clears all notifications
On FILTER click - shows the pop menu with a filter options

1. I am not sure about the UI. Maybe better to replace text with icons? Any other suggestions?
2. Current implementation requires to click "Apply" to apply filter to notifications. Not sure what is better - hide the popup menu and apply filter on any items click (the standard popup menu behaviour) or leave it as implemented now (with an "Apply" button). Maybe be better to replace it with a popup window? If current implementation is ok for you - i want to add a bold for the "Apply" item (with Spannable). Maybe add add a divider above it would be ok too. Any other suggestions?
3. App saves the selected filter at the default shared preferences. Is it ok?

### CODE REVIEW

File: app/src/main/res/layout/item_notifications_top.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (connyduck) [inline app/src/main/res/layout/item_notifications_top.xml]: a `Linear Layout` should be enough

File: app/src/main/java/com/keylesspalace/tusky/fragment/NotificationsFragment.java [hunk 2]
```diff
@@ -545,6 +565,161 @@ public void onNotificationContentCollapsedChange(boolean isCollapsed, int positi
...
+        String json = preferences.getString("notifications.filter", "[]");
+        Set<Notification.Type> set = getNotificationsGson().fromJson(json, new TypeToken<Set<Notification.Type>>() {
+        }.getType());
+        if (set != null)
+            notificationFilter.addAll(set);
+    }
+
+    private void saveNotificationsFilter() {
+        String json = getNotificationsGson().toJson(notificationFilter, new TypeToken<Set<Notification.Type>>() {
+        }.getType());
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/fragment/NotificationsFragment.java]: please de/serialize the settings in a more efficient way, e.g. by concatenating the active ones to a string seperated by some special character. Gson is way too expensive for this simple task.

**Comment by pandasoft0** [comment]
@connyduck Could you please check new update?

**Comment by connyduck** [comment]
As I wrote in the dev chat, I am very busy right now and have no time to check PRs. But don't worry, I won't forget it.

File: app/src/main/java/com/keylesspalace/tusky/util/NotificationTypeConverter.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package com.keylesspalace.tusky.util
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/util/NotificationTypeConverter.kt]: please add the License information at the top of the file

**Comment by ralexey** [comment]
Ok, let me fix the issue with scrolling before merge. I will try to do it today.

**Comment by pandasoft0** [comment]
Let me fix the issue with scroll on empty screen before merge. I will do it today.

**Comment by pandasoft0** [comment]
Done

**Pull request merged**